Update tools-deps to 1.10.3.855

imrekoszo opened the pull request:
1.10.3.855 is the current stable version: https://github.com/clojure/brew-install/blob/1.10.3/CHANGELOG.md

It includes the changes from 1.10.3.849 which addresses some CVE security concerns: https://clojure.org/releases/tools#v1.10.3.855

I also reverted an earlier commit of mine after discovering that there are wrapper shell scripts for arguments to `docker-clojure.core/-main`, sorry about that.

cap10morgan commented:
Thanks @imrekoszo! I have this in progress locally already. I’m just traveling for work and so it’s taking me a little longer than usual. I’ll get it updated today.

cap10morgan commented:
@imrekoszo Out of curiosity, why did you revert the argument passthrough?

imrekoszo commented:
That's grand, thank you. I see it's now updated.

> why did you revert the argument passthrough?

I suppose I'll first answer why I had added it in the first place; I wanted to use the `clean` functionality from `-main` and did not notice there was `clean.sh` wrapping that already, I only saw `build-images.sh` and wanted to say `./build-images.sh clean`.

```clojure
(defn -main [& args]
  (case (first args)
    "clean" (df/clean-all)
    "dockerfiles" (dorun (generate-dockerfiles!))
    (build-images (generate-dockerfiles!)))
  (System/exit 0))
```

Once I realised there were separate scripts wrapping the 3 cases there, I thought that my original change was unnecessary. That's why I decided to revert it.

cap10morgan commented:
> Once I realised there were separate scripts wrapping the 3 cases there, I thought that my original change was unnecessary. That's why I decided to revert it.

I see, that makes sense. I don't totally love our separate shell scripts approach and have played around with some alternatives, but nothing mergeable has emerged yet.

I hadn't noticed that clean.sh needed updating yet either, so thanks for catching that. I included it in my PR today.

imrekoszo commented:
In many projects we use a Justfile as a frontend to the most common tasks. Makes the entry barrier a bit lower if you can ask the project what you can do with it. Babashka also added tasks recently which look very similar. The downside of both is that you need to have just or bb installed to interact with them, so they might not be a good solution for this project.
